### PR TITLE
Add bids validation to integration tests

### PR DESCRIPTION
**Pull request by fordmcdonald**
The main task of this PR was to add bids validation into our integration tests. I created a small utility function `validateBIDS()`, which writes the bids-validator output to a file (issues.txt).  From there, the function searches for "ERR" flags line by line, asserting that no errors have been raised. 

You'll notice I have added `skiplist = ["6"]` to many of the xnat2bids parameter lists.  This is because the current BIDS specification does not support multiple echo T1w scans (https://github.com/bids-standard/bids-specification/issues/654).

### CODE REVIEW

File: tests/integration/test_postprocessing.py [hunk 1]
```diff
@@ -293,3 +298,13 @@ def check_json_data(fieldmaps: list, op: str, value: str):
             data = json.load(f)
             assert rel_ops[op](data["IntendedFor"], value)
             f.close
+
+
+def validateBIDS(bids_root_dir: str):
```
Comment (mirestrepo) [inline tests/integration/test_postprocessing.py:303]: so far `bids-validator` wasn't a dependency of xnat-tools. This makes it one. How are you envisioning that?

**Comment by fordmcdonald** [comment]
@mirestrepo I verified BIDS validation passed for multiple sessions of a given subject.  We can expand this test further to be more exhaustive across DEMODAT.  To trigger the job on  a merge to main, I added the following line `if: ${{ github.ref == 'refs/heads/main' }}` to the bids-validation job.  You can see above that the "Run system test for bids-validator" job was skipped on pull request.  Once you approve these changes, I can merge to main and validate the bids-validator job was successfully triggered.

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,15 @@
-FROM python:3.10
+FROM python:3.10.6
 
 ENV POETRY_VERSION=1.0.3
 ENV DCM2NIIX_VERSION=v1.0.20190902
 
+# Install Node.js and npm
+RUN curl -fsSL https://deb.nodesource.com/setup_lts.x | bash - \
```
Comment (mirestrepo) [inline Dockerfile:7]: We are installing node in the action - so we don't need it baked into the docker image as well

File: Dockerfile [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+    && apt-get install -y nodejs
+
+# Install bids-validator using npm
+RUN npm install -g bids-validator
```
Comment (mirestrepo) [inline Dockerfile:11]: We are installing bids-validator in the action. We don't need it baked into the container.

We'll only need to do this if the action doesn't work

**Review by mirestrepo — CHANGES_REQUESTED** [review]
This is basically done - but since we decided to install node and bids validator in the action (so only temporary in the runner), we don't need it in the docker image.

**Review by mirestrepo — APPROVED** [review]
Changes look good. 
You could though before merging, removing the `if: ${{ github.ref == 'refs/heads/main' }}`.. just to check that BIDS validation works... (and temporary you could add it to the tests to skip those) because otherwise, you'll need to fix the action by pushing to main which is more cumbersome. Once that passes, you can bring your statements back in order. I'm just not sure the setup node part will workout smoothly

**Pull request merged**